src/main.cpp

```diff
@@ -1933,249 +1936,144 @@ void MainFrame::destroy_src(void)
+        
+        auto engine = AudioEngineFactory::GetAudioEngine();
+        engine->setOnEngineError([&](IAudioEngine&, std::string error, void* state) {
+            wxMessageBox(wxString::Format(
+                         "Error encountered while initializing the audio engine: %s.", 
+                         error), wxT("Error"), wxOK, this); 
+        }, nullptr);
+        engine->start();
+
+        if (g_nSoundCards == 0) 
         {
-            wxMessageBox(wxT("Port Audio failed to initialize"), wxT("Pa_Initialize"), wxOK);
-        }
-
-        m_rxInPa = new PortAudioWrap();
-        if(g_soundCard1InDeviceNum != g_soundCard1OutDeviceNum)
-            two_rx=true;
-        if(g_soundCard2InDeviceNum != g_soundCard2OutDeviceNum)
-            two_tx=true;
-
-        if (g_verbose) fprintf(stderr, "two_rx: %d two_tx: %d\n", two_rx, two_tx);
-        if(two_rx)
-            m_rxOutPa = new PortAudioWrap();
-        else
-            m_rxOutPa = m_rxInPa;
-
-        if (g_nSoundCards == 0) {
             wxMessageBox(wxT("No Sound Cards configured, use Tools - Audio Config to configure"), wxT("Error"), wxOK);
-            delete m_rxInPa;
-            if(two_rx)
-                delete m_rxOutPa;
-            m_RxRunning = false;
-            Pa_Terminate();
-            return;
-        }
-
-        // Init Sound card 1 ----------------------------------------------
-        // sanity check on sound card device numbers
 
-        if ((m_rxInPa->getDeviceCount() <= g_soundCard1InDeviceNum) ||
-            (m_rxOutPa->getDeviceCount() <= g_soundCard1OutDeviceNum)) {
-            wxMessageBox(wxT("Sound Card 1 not present"), wxT("Error"), wxOK);
-            delete m_rxInPa;
-            if(two_rx)
-                delete m_rxOutPa;
             m_RxRunning = false;
-            Pa_Terminate();
-            return;
-        }
-
-        // work out how many input channels this device supports.
-
-        deviceInfo1a = Pa_GetDeviceInfo(g_soundCard1InDeviceNum);
-        if (deviceInfo1a == NULL) {
-            wxMessageBox(wxT("Couldn't get input device info from Port Audio for Sound Card 1"), wxT("Error"), wxOK);
-            delete m_rxInPa;
-            if(two_rx)
-				delete m_rxOutPa;
-            m_RxRunning = false;
-            Pa_Terminate();
-            return;
-        }
-        if (deviceInfo1a->maxInputChannels == 1)
-            inputChannels1 = 1;
-        else
-            inputChannels1 = 2;
-
-        // Grab the info for the FreeDV->speaker device as well and ensure we're using
-        // the smallest number of common channels (1 or 2).
-        deviceInfo1b = Pa_GetDeviceInfo(g_soundCard1OutDeviceNum);
-        if (deviceInfo1b == NULL) {
-            wxMessageBox(wxT("Couldn't get output device info from Port Audio for Sound Card 1"), wxT("Error"), wxOK);
-            delete m_rxInPa;
-            if(two_rx)
-				delete m_rxOutPa;
-            m_RxRunning = false;
-            Pa_Terminate();
+            
```

Comment:
So nice to clean all this up :+1:  I really was "feeling my way" when I wrote this (about 10 years ago now).